client2/src/components/game/game-renderer.tsx

```diff
@@ -32,43 +32,38 @@ export const GameRenderer: React.FC = () => {
         elem.getContext('2d')?.scale(cst.TILE_RESOLUTION, cst.TILE_RESOLUTION)
     }
 
-    React.useEffect(() => {
+    // Since this is a callback, we need to ensure we recreate the function when the
+    // active match changes. Similarly, the event listener needs to be updated, which
```

Comment:
Is this happening automatically? Having activeMatch in callback dependencies should do this right?

Reply:
Yeah, this is automatic here